# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: tools/profiler/core/ProfileBuffer.h
# Commit: b6d73f5042a7
# Full Hash: b6d73f5042a7f3703b5d16e222e9f71153a43164
# Author: Gerald Squelart <gsquelart@mozilla.com>
# Date: 2019-09-17 15:54:53
# Regressor Bug: 1576551
# File Overlap Count: 1
# Description:
#   Bug 1576551 - Store BlocksRingBuffer outside of ProfileBuffer - r=gregtatum
#   
#   The main `BlocksRingBuffer`s will be stored in `CorePS` (outside of
#   `ProfileBuffer`s), as we need to be able to safely access the underlying buffers
#   when profilers are not enabled.
# ==============================================================================

diff -r 366030bd2d84 -r b6d73f5042a7 tools/profiler/core/ProfileBuffer.h
--- a/tools/profiler/core/ProfileBuffer.h	Tue Sep 17 01:50:19 2019 +0000
+++ b/tools/profiler/core/ProfileBuffer.h	Tue Sep 17 02:12:42 2019 +0000
@@ -27,8 +27,10 @@
   using BlockIndex = mozilla::BlocksRingBuffer::BlockIndex;
 
   // ProfileBuffer constructor
+  // @param aBuffer The BlocksRingBuffer to use as buffer manager.
   // @param aCapacity The capacity of the buffer.
-  explicit ProfileBuffer(mozilla::PowerOfTwo32 aCapacity);
+  ProfileBuffer(mozilla::BlocksRingBuffer& aBuffer,
+                mozilla::PowerOfTwo32 aCapacity);
 
   ~ProfileBuffer();
 
@@ -144,7 +146,7 @@
       mozilla::BlocksRingBuffer& aBlocksRingBuffer, int aThreadId);
 
   // The circular-ring storage in which this ProfileBuffer stores its data.
-  mozilla::BlocksRingBuffer mEntries;
+  mozilla::BlocksRingBuffer& mEntries;
 
  public:
   // `BufferRangeStart()` and `BufferRangeEnd()` return `uint64_t` values